src/cobra/io/web/load.py

```diff
@@ -0,0 +1,201 @@
+"""Provide a function ``load_model`` to access remote model repositories."""
+
+
+import gzip
+import logging
+import pathlib
+from typing import TYPE_CHECKING, Iterable, Optional, Union
+
+import appdirs
+import diskcache
+import httpx
+import libsbml
+
+from ..sbml import _sbml_to_model
+from .abstract_model_repository import AbstractModelRepository
+from .bigg_models_repository import BiGGModels
+from .biomodels_repository import BioModels
+
+
+if TYPE_CHECKING:
+    from cobra.core import Model
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_model(
+    model_id: str,
+    repositories: Iterable[AbstractModelRepository] = (BiGGModels(), BioModels()),
+    cache: bool = True,
+    cache_directory: Optional[Union[pathlib.Path, str]] = None,
+) -> "Model":
+    """
+    Download an SBML model from a remote repository.
+
+    Downloaded SBML documents are by default stored in a cache on disk such that future
+    access is much faster. By default, models can be loaded from the following
+    repositories:
+
+    * BiGG Models
+    * BioModels
+
+    You can use the ``AbstractModelRepository`` class as a parent to implement your own
+    repository accessor which you pass to the ``load_model`` function. In case you
+    implement a new interface, please consider submitting a pull request to COBRApy.
+
+    Parameters
+    ----------
+    model_id : str
+        The identifier of the desired metabolic model. This is typically repository
+        specific.
+    repositories : iterable, optional
+        An iterable of repository accessor instances. The model_id is searched in order.
+    cache : bool, optional
+        Whether or not to use the local caching mechanism (default yes).
+    cache_directory : pathlib.Path or str, optional
```

Comment:
Set a maximum size for this directory. For instance a lot of my workflows download hundreds or thousands of models. I would not want cobrapy using GBs of storage that I can't control.

Reply:
You can disable the caching entirely, of course, but I will add it to the configuration.